src/UiPath.CoreIpc.Tests/NamedPipeTests.cs

```diff
@@ -43,6 +43,29 @@ public async Task PipeSecurityForWindows()
             _ = protectedService.RunAsync();
             await CreateSystemService().DoNothing().ShouldThrowAsync<UnauthorizedAccessException>();
         }
+
+        [Fact]
+        public async Task PipeCancelIoOnServer()
+        {
+            (await _systemClient.CancelIoPipe(new())).ShouldBeTrue();
+
+            (await _systemClient.Delay()).ShouldBeTrue();
+        }
+
+        [Fact]
+        public async Task PipeCancelIoOnClient()
```

Comment:
I think we're not covering some parts of the complexities in `Helpers.cs`. Maybe we would benefit from 2 more tests similar to this one, which could:

1 - call `CancelIOEx` twice **tightly** and expect an exception
2 - call `CancelIOEx` twice **with more than 1 second in between** and expect that everything repairs itself

Reply:
Done